src/components/fields/switch/switch.tsx

```diff
@@ -0,0 +1,94 @@
+import { Form } from "@lifesg/react-design-system/form";
+import { Toggle } from "@lifesg/react-design-system/toggle";
+import { useEffect, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import * as Yup from "yup";
+import { TestHelper } from "../../../utils";
+import { useValidationConfig } from "../../../utils/hooks";
+import { IGenericFieldProps } from "../../frontend-engine";
+import { FlexWrapper } from "./switch.styles";
+import { ISwitchSchema } from "./types";
+
+export const Switch = (props: IGenericFieldProps<ISwitchSchema>) => {
+	// =============================================================================
+	// CONST, STATE, REFS
+	// =============================================================================
+	const {
+		schema: { label, disabled, validation, customOptions, ...otherSchema },
+		id,
+		value,
+		error,
+		onChange,
+	} = props;
+
+	const [stateValue, setStateValue] = useState<boolean>(value || undefined);
+	const { setFieldValidationConfig } = useValidationConfig();
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		setFieldValidationConfig(id, Yup.boolean(), validation);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [validation]);
+
+	useEffect(() => {
+		setStateValue(value);
+	}, [value]);
+
+	// =============================================================================
+	// EVENT HANDLERS
+	// =============================================================================
+	const handleClick = (value: boolean) => {
+		//console.log("click", stateValue, value);
+		onChange({ target: { value } });
+	};
+
+	// =============================================================================
+	// HELPER FUNCTIONS
+	// =============================================================================
+	const isSwitchChecked = (value: boolean) => {
+		return stateValue === value;
+	};
+
+	const formatId = (type: string) => {
+		return `${id}-${type}`;
+	};
+
+	return (
+		<Form.CustomField id={id} label={label} errorMessage={error?.message}>
+			<FlexWrapper>
```

Comment:
can i have a reference of similar implementation for this on any FEE component? As aria-label is currently only used in image components.

Reply:
hmm I don't have a good example in FEE. to be more specific, set `role="radiogroup"` and `aria-label={label}` on FlexWrapper

this makes it so that screenreaders will read out the label when the user focuses on one of the radio buttons

Reply:
okay added those on the FlexWrapper